samples/counting_sort.wgsl

```diff
@@ -0,0 +1,82 @@
+/*
+ * Copyright 2025 The wgsl-fuzz Project Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+const ARRAY_LENGTH = 16;
+
+const I32_MIN = -2147483648;
+const MAX_ELEMENT_UPPER_BOUND = 256;
+
+struct Number {
+    @size(16) value: i32
```

Comment:
At the moment, the transformation framework isn't going to do the right thing with the use of `@size(16)` here. While it would be good to sort that out in due course (can you open an issue?) I think that in the interests of having usable samples sooner rather than later it would be good to address this using padding.

I suggest that you give the `Number` struct additional fields that pad it out to 16 bytes. Maybe it would be interesting to do this via a single `vec2` field called `unused_padding`. That would be interesting becuase even though the struct would only have 12 bytes' worth of actual data, because the `unused_padding` field would be aligned to 8 bytes the overall struct size will be 16.